### PR TITLE
Use checkout@3.5.3 version to pull correct code tag, not master

### PR DESCRIPTION
**Pull request by uncleDecart**
Following on #910 we saw that EVE checks out Eden master code and not code from tag from yml file and if master fails tests are failing on EVE, which is undesirable

https://github.com/lf-edge/eve/actions/runs/6613253197/job/17960730583?pr=3516#step:2:474

Investigating it more turns out in version 3.5.3 of github checkout action behaviour is what we need. Check #909 for more info

### CODE REVIEW

**Comment by uncleDecart** [comment]
We need to backport this to 0.9.2 so EVE tests will not fail

**Review by giggsoff — APPROVED** [review]
Thank you for find it out.
Can we somehow check the changes before merging and re-tagging?

**Comment by uncleDecart** [comment]
Well, we can rebase swithing new workflow PR to this and see if VNC test passes

**Comment by giggsoff** [comment]
> Well, we can rebase swithing new workflow PR to this and see if VNC test passes

As I understand the main value of thos PR is for EVE-OS repo. My question was about that case.

**Comment by uncleDecart** [comment]
Well backporting to 0.9.2 should do the trick. Releasing 0.9.3 will work only if we either revert cloud-init test, point to eve hash which has the support, wait for next release which includes it

**Comment by giggsoff** [comment]
> Well backporting to 0.9.2 should do the trick. Releasing 0.9.3 will work only if we either revert cloud-init test, point to eve hash which has the support, wait for next release which includes it

I am voiting for pointing onto eve hash (I am not sure about EVE-OS release plan). Re-assing of tags is not the best approach.

Another question: for lts relases of EVE-OS in EVE-OS repo (for pushing patch tags for old branches), which tag of new Eden workflow is in use?

**Comment by uncleDecart** [comment]
Right now 0.9.2 is used with new workflows, where they are used. So in LTS with new workflows it'll be 0.9.2 that's why I suggested to merge this patch in 0.9.2 tag, so stable branches won't be affected

**Comment by giggsoff** [comment]
I still cannot understand, sorry. Can you please describe a plan? You wrote, that you want to remove and re-apply the tag to the master branch with the PR merged, correct? But how about stable branches of EVE-OS (you mentioned that they use 0.9.2 tag of Eden) without support for new cloud-init test? Looks like they will be affected by novel test they aren't support, but will see in 0.9.2 tag.

**Comment by europaul** [comment]
just for future reference: bumping up the version sadly didn't solve the problem and it was necessary to add and `eden_version parameter` as input to the workflow in https://github.com/lf-edge/eden/pull/961

**Pull request merged**
